solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/EuristicBoostedModel.java

```diff
@@ -0,0 +1,150 @@
+    public static final String FEATURE_NAME = "feature";
+    public static final String WEIGHT = "weight";
+    public static final String BOOST_TYPE = "type";
+
+    public String feature;
+    public int featureIndex;
+    public float weight;
+    public String type;
+
+    public EuristicFeatureBoost(Map<String, Object> featureBoostParam) {
+      checkParams(featureBoostParam);
+
+      feature = ((String) featureBoostParam.get(FEATURE_NAME));
+      weight = NamedParams.convertToFloat(featureBoostParam.get(WEIGHT));
+      featureIndex = getFeatureIndex(feature);
+      type = MULTIPLICATIVE_BOOST;
+      if (featureBoostParam.get(BOOST_TYPE) != null) {
+        type = (String) featureBoostParam.get(BOOST_TYPE);
+      }
+    }
+
+    private void checkParams(Map<String, Object> euristicBoostParams) {
+      checkParam(euristicBoostParams, FEATURE_NAME);
+      checkParam(euristicBoostParams, WEIGHT);
+    }
+
+    private void checkParam(Map<String, Object> featureBoostParams, String paramName) {
+      String modelName=EuristicBoostedModel.this.getName();
+      if (!featureBoostParams.containsKey(paramName)) {
+        throw new ModelException("Model " + modelName + " required param " + paramName + " not defined");
+      }
+    }
+
+    private int getFeatureIndex( String featureName) {
+      String modelName=EuristicBoostedModel.this.getName();
+      final List<Feature> features = getFeatures();
+      Optional<Feature> featureToBoostOptional = features.stream()
+          .filter(f -> f.getName().equals(featureName))
+          .findFirst();
+
+      if (featureToBoostOptional.isPresent()) {
+        Feature featureToBoost = featureToBoostOptional.get();
+        return featureToBoost.getId();
+      } else {
+        throw new ModelException("Model " + modelName + " doesn't contain any feature with the name=[" + feature + "}");
+      }
+    }
+  }
+
+  public EuristicBoostedModel(String name, List<Feature> features,
+                              String featureStoreName, List<Feature> allFeatures,
+                              NamedParams params) throws ModelException {
+    super(name, features, featureStoreName, allFeatures, params);
+    initEuristicBoost();
+  }
+
+  protected void initEuristicBoost() {
+    Map<String, Object> params = (Map<String, Object>) EuristicBoostedModel.this.getParams().get(EURISTIC_BOOST_PARAM);
+    if (params != null) {
+      this.euristicFeatureBoost = new EuristicFeatureBoost(params);
+    }
+  }
+
+  @Override
+  public float score(float[] modelFeatureValuesNormalized) {
+    internalModelScore = LTRModel.score(modelFeatureValuesNormalized);
+    float finalScore = internalModelScore;
+    if (euristicFeatureBoost != null) {
+      boostFeatureValue = modelFeatureValuesNormalized[euristicFeatureBoost.featureIndex];
+      weightedBoostFeatureValue = euristicFeatureBoost.weight * boostFeatureValue;
+
+      if (MULTIPLICATIVE_BOOST.equals(euristicFeatureBoost.type)) {
```

Comment:
... here: string comparison at scoring time. if you use an enum we can also use a switch case here.